tests/builders/test_rnn.py

```diff
@@ -49,44 +52,65 @@ def rnn_match_cfg(rnn: RNN, rnn_cfg: rnn_pb2.RNN, input_features: int) -> None:
         bias += getattr(rnn.rnn, f"bias_hh_l{l}")[
             hidden_size : 2 * hidden_size
         ]
-        assert torch.allclose(bias, torch.tensor(forget_gate_bias))
+        assert torch.allclose(
+            bias, torch.tensor(forget_gate_bias).to(bias.device)
+        )
 
 
 @st.composite
 def rnn_cfg_tensors(
     draw,
-) -> st.SearchStrategy[Tuple[torch.nn.Module, rnn_pb2.RNN, torch.Tensor]]:
+) -> st.SearchStrategy[
+    Tuple[torch.nn.Module, rnn_pb2.RNN, torch.Tensor, bool]
+]:
     """Returns a search strategy for RNNs built from a config + valid input."""
+    batch_first = draw(st.booleans())
     rnn_cfg = draw(rnns())
     tensor = draw(tensors(min_n_dims=3, max_n_dims=3))
-    rnn, _ = build(rnn_cfg, input_features=tensor.size(2))
-    return rnn, rnn_cfg, tensor
+    if batch_first:
+        tensor = tensor.transpose(1, 0)
+    rnn, _ = build(
+        rnn_cfg, input_features=tensor.size(2), batch_first=batch_first
+    )
+    return rnn, rnn_cfg, tensor, batch_first
 
 
 # Tests -----------------------------------------------------------------------
 
 
-@given(rnn_cfg=rnns(), input_features=st.integers(1, 128))
+@settings(deadline=3000)
+@given(
+    rnn_cfg=rnns(),
+    input_features=st.integers(1, 32),
+    batch_first=st.booleans(),
+)
 def test_build_rnn_returns_correct_rnn_with_valid_params(
-    rnn_cfg: rnn_pb2.RNN, input_features: int
+    rnn_cfg: rnn_pb2.RNN, input_features: int, batch_first: bool
 ) -> None:
     """Test that build_rnn returns the correct RNN with valid params."""
-    rnn, rnn_output_size = build(rnn_cfg, input_features)
-    rnn_match_cfg(rnn, rnn_cfg, input_features)
+    rnn, rnn_output_size = build(rnn_cfg, input_features, batch_first)
+    rnn_match_cfg(rnn, rnn_cfg, input_features, batch_first)
 
 
 @given(rnn_cfg_tensor=rnn_cfg_tensors())
 def test_build_rnn_rnn_forward_output_correct_size(
-    rnn_cfg_tensor: Tuple[torch.nn.Module, rnn_pb2.RNN, torch.Tensor]
+    rnn_cfg_tensor: Tuple[torch.nn.Module, rnn_pb2.RNN, torch.Tensor, bool]
 ) -> None:
     """Ensures returned RNN forward produces output with correct size."""
-    rnn, rnn_cfg, tensor = rnn_cfg_tensor
-    seq_len, batch, input_features = tensor.size()
+    rnn, rnn_cfg, tensor, batch_first = rnn_cfg_tensor
+    if not batch_first:
```

Comment:
Since you have an if and else branch anyway, this negation is redundant (i.e. make the condition just `if batch_first` and swap the branches). I think it also makes it a bit nicer to read. Same applies to the if statement below.

Reply:
Good point thanks I've updated!